File: programmers/2022 KAKAO BLIND RECRUITMENT/주차 요금 계산.js
function timeToMinute(time) {
  const [h, m] = time.split(":").map((x) => parseInt(x));
  return h * 60 + m;
}

function solution(fees, records) {
  const cars = {};
  records.forEach((r) => {
    const [time, car, inOut] = r.split(" ");
    const t = timeToMinute(time);
    if (car in cars) {
      cars[car].push([t, inOut]);
    } else {
      cars[car] = [[t, inOut]];
    }
  });
  const orders = [];
  for (const [car, times] of Object.entries(cars)) {
    let t = 0;
    for (const time of times) {
      if (time[1] == "IN") {
        t -= time[0];
      } else {
        t += time[0];
      }
    }
    if (t <= 0) {
      t += 23 * 60 + 59;
    }
    orders.push([car, t]);
  }
  const ans = orders
    .sort((a, b) => a[0] - b[0])
    .map((data) => {
      let fee = fees[1];
      const t = data[1];
      if (t > fees[0]) {
        fee += Math.ceil((t - fees[0]) / fees[2]) * fees[3];
      }
      return fee;
    });
  return ans;
}

File: programmers/2022 KAKAO BLIND RECRUITMENT/주차 요금 계산.py
import math


def time_to_minute(time):
    h, m = map(int, time.split(':'))
    return h * 60 + m
    

def solution(fees, records):
    ans = []
    cars = dict()
    for record in records:
        time, car, in_out = record.split()
        if car in cars:
            cars[car].append([time_to_minute(time), in_out])
        else:
            cars[car] = [[time_to_minute(time), in_out]]
    cars = sorted(cars.items())
    for car in cars:
        t = 0
        for r in car[1]:
            if r[1] == 'IN':
                t -= r[0]
            else:
                t += r[0]
        if car[1][-1][1] == 'IN':
            t += time_to_minute('23:59')
        fee = fees[1]
        if t > fees[0]:
            fee += math.ceil((t - fees[0]) / fees[2]) * fees[3]
        ans.append(fee)
    return ans